notation.go

```diff
@@ -203,6 +206,18 @@ func Verify(ctx context.Context, verifier Verifier, repo registry.Repository, re
 
 			// verify each signature
 			outcome, err := verifier.Verify(ctx, artifactDescriptor, sigBlob, opts)
+			// process verification items
+			for _, result := range outcome.VerificationResults {
```

Comment:
We probably don't need to exeute this bock when `--debub` is not enabled. Can we use context to pass that information to all functions?

Reply:
1. I think notation-go should not deal with CLI flag. The logic should handled in CLI to enable or diable the log.
2. We should reach to a consensus. We prefer to show the warning and error log even if the `--debug` is disabled, because according to our trust policy spec, if the verification level is audit or the action is logger, we should always log the error.
@shizhMSFT @yizha1

Reply:
Update:
- According to [trust policy spec](https://github.com/notaryproject/notaryproject/blob/main/specs/trust-store-trust-policy.md#signature-verification-details) we should always show the verification action "logged" error. From the perspective of notation-go we should log the errors as it defined in spec.
- When should we show the log?
From the perspective of user, they may do not need the log, however, we didn't define when should we enable or disable the log based on our spec. According to current spec, we should always log verification action "logged" error.

So can we keep showing the warning & error logs even if without `--debug` flag for RC.1, and let's update the spec and CLI behavior after we receive feedbacks from users.

Reply:
> According to [trust policy spec](https://github.com/notaryproject/notaryproject/blob/main/specs/trust-store-trust-policy.md#signature-verification-details) we should always show the verification action "logged" error. From the perspective of notation-go we should log the errors as it defined in spec.
That's a really good point, if we are evaluating a single signature it makes sense to display this information. But in a scenario where there are 10 signatures, and signature verification succeeds for the 10th signature, if we display information for all signatures we will overload the user with information that they don't even care about. Probably the right balance would be to display this information only for the signature for which verification was successful.